server/controllers/items/lib/bulk_update.js

```diff
@@ -0,0 +1,45 @@
+const items_ = require('controllers/items/lib/items')
+const Item = require('models/item')
+const { emit } = require('lib/radio')
+const { partition } = require('lodash')
+const _ = require('builders/utils')
+const { validateVisibilityKeys } = require('lib/visibility/visibility')
+const { validateShelves } = require('controllers/items/lib/validate_item_async')
+
+const bulkItemsUpdate = async ({ reqUserId, ids, attribute, value, attempt = 0, previousUpdates = [] }) => {
+  const itemUpdateData = { [attribute]: value }
+  const currentItems = await items_.byIds(ids)
+  const formattedItems = currentItems.map(currentItem => Item.update(reqUserId, itemUpdateData, currentItem))
+  await validateValue({ attribute, value, reqUserId })
+  try {
+    const successfulUpdates = await items_.db.bulk(formattedItems)
+    await emit('user:inventory:update', reqUserId)
+    return previousUpdates.concat(successfulUpdates)
+  } catch (err) {
+    if (attempt > 10) throw err
+    const { body } = err.context
+    const [ failedUpdates, successfulUpdates ] = partition(body, hasError)
+    _.warn({ failedUpdates, successfulUpdates, attempt }, 'retrying bulk items update')
+    return bulkItemsUpdate({
+      reqUserId,
+      ids: failedUpdates.map(getId),
+      attribute,
```

Comment:
Documenting "why is `attribute` representing" could be helpful, since the below `visibility` and `shelves` refers to very different concept.

Reply:
both `visibility` and `shelves` arrays are attributes of the item doc, so that seemed self-explanatory enough to me, but if you can come up with something that addresses that issue, that would work for me too